File: 0242-valid-anagram/0242-valid-anagram.cpp
// time:o(s+t)
// space:o(s+t)
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        
        unordered_map<char,int> smap;
        unordered_map<char,int> tmap;
        
        for(int i=0;i<t.size();i++){
            smap[s[i]]++;
            tmap[t[i]]++;
            
            }
        return smap == tmap;
        
        
    }
};

/*
int main() {
    string input1, input2;
    cout << "Enter the first string: ";
    getline(cin, input1);
    cout << "Enter the second string: ";
    getline(cin, input2);

    Solution solution;
    bool result = solution.isAnagram(input1, input2);

    cout << "Are the strings anagrams? " << (result ? "Yes" : "No") << endl;

    return 0;
}
*/

File: 0242-valid-anagram/0242-valid-anagram.py
## time:o(s+t)
## space:o(s+t)
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if(len(s) != len(t)):
            return False;
        
        mapS={}
        mapT={}
        
        for i in range(len(s)):
            mapS[s[i]] = 1 + mapS.get(s[i],0)
            mapT[t[i]] = 1 + mapT.get(t[i],0)
            
        return mapS == mapT 
    
    
# Get user input for s and t
##s = input("Enter the first string: ")
##t = input("Enter the second string: ")

##solution = Solution()
##print(solution.isAnagram(s, t))    
        